### ISSUE TITLE
deprication notice while build: Node#moveTo was deprecated. Use Container#append.

### ISSUE THREAD

**Issue by Sibiraj-S** [angular/angular-cli#8598]
#### Version

```

    _                      _                 ____ _     ___
   / \   _ __   __ _ _   _| | __ _ _ __     / ___| |   |_ _|
  / △ \ | '_ \ / _` | | | | |/ _` | '__|   | |   | |    | |
 / ___ \| | | | (_| | |_| | | (_| | |      | |___| |___ | |
/_/   \_\_| |_|\__, |\__,_|_|\__,_|_|       \____|_____|___|
               |___/
    
Angular CLI: 1.5.3
Node: 8.9.1
OS: linux x64
Angular: 5.0.0
... animations, common, compiler, compiler-cli, core, forms
... http, language-service, platform-browser
... platform-browser-dynamic, router

@angular/cli: 1.5.3
@angular/tsc-wrapped: 4.4.6
@angular-devkit/build-optimizer: 0.0.32
@angular-devkit/core: 0.0.20
@angular-devkit/schematics: 0.0.35
@ngtools/json-schema: 1.1.0
@ngtools/webpack: 1.8.3
@schematics/angular: 0.1.0
typescript: 2.4.2
webpack: 3.8.1
```

### Observed behavior
```
30% building modules 167/168 modules 1 active ...nts/src/styles.scssNode#moveTo was deprecated. Use Container#append.
```
After updating to 1.5.3 I get this notice with `scss` while running `ng build --prod`

**Comment by filipesilva** [comment]
Heya, I tried this on a new project with this `src/styles.scss`:
```
body {
  background-color: #000000
}
```
But couldn't see that message. 

This seems like a bug but we'll need to look at a reproduction to find and fix the problem. Can you setup a minimal repro please? 

You can read [here](https://github.com/angular/angular-cli/blob/master/CONTRIBUTING.md#-submitting-an-issue) why this is needed. A good way to make a minimal repro is to create a new app via `ng new repro-app` and adding the minimum possible code to show the problem. Then you can push this repository to github and link it here.

I guess it's related to some dependency update but there might be more to it.

**Comment by clydin** [comment]
This is coming from cssnano.  It still uses some deprecated methods.  The methods still function and the warning will only be shown at most once.  The next version of cssnano (when released) should completely resolve the issue.

**Comment by Sibiraj-S** [comment]
sure @filipesilva.

**Comment by Sibiraj-S** [comment]
@filipesilva .

#### Steps to Reproduce. 

This is what I  did. I set default stylesheets to `scss`

1. ng new myApp
2. npm install bootstrap@next
3. import bootstrap.scss into style.scss
```
/* You can add global styles to this file, and also import other style files */

@import "../node_modules/bootstrap/scss/bootstrap"; // I used bootstrap

```
4. ng build -- prod

The problem occurs while importing files into styles.scss

**Comment by gpit2286** [comment]
Some issue when trying to import normalize.css. 

Same setup. SCSS, @import, etc.

**Comment by dragonflypl** [comment]
Same issue here: `Build stderr: Node#moveTo was deprecated. Use Container#append.`

```
"styles": [
        "path-to-bootstrap4-sass-here"
      ],
```

**Comment by Dok11** [comment]
I have same trouble with bootstrap 4 file `_normalize.scss` for rule:
```scss
b,
strong {
  font-weight: inherit;
}
```

![image](https://user-images.githubusercontent.com/2697890/34524284-042f8c0a-f0ac-11e7-81b7-2fe07191b6cb.png)

but without `b` all is fine:
![image](https://user-images.githubusercontent.com/2697890/34524317-313d2324-f0ac-11e7-845e-a5066e75b025.png)

**Comment by skorunka** [comment]
Same problem, cli 1.6.3

**Comment by fusionx86** [comment]
I'm having problems with this too and it's causing TeamCity builds to fail that are otherwise clean. I haven't found a good workaround for this, but the real problem is non-error messages being sent to STDERR. When we will have the ability to filter non-error messages? Even better would be to stop sending everything to the wrong output. Not sure how this behavior is acceptable...

**Comment by johnlacr** [comment]
Most likely the node.js deprecate method in the util method is being used - this is documented to write to STDERR: https://nodejs.org/api/util.html#util_util_deprecate_function_string

**Comment by polyglotinc** [comment]
same problem with cli 1.6.6

**Comment by Londovir** [comment]
This one is getting frustrating on our TFS/MSBuild chains because it throws a "partial success" (or partial fail, depending on your viewpoint), which makes it look like our build is faulty. I've tried using variations on the --no-deprecation flag for Node, but no way of launching my scripts seems to make that flag work to hide the deprecation alert. I've even tried editing an ejected webpack.config.js file to include a `process.noDeprecation = true;` line, but that doesn't work either. Is there no way to get around the spurious deprecation warning until 1.7.0 comes out someday?

**Comment by ubarevicius** [comment]
I'm experiencing the same problem, spent an hour to try and silence this on our build server with no luck. cli v1.6.5. I don't see it on cli 1.7.0 beta 2, but ng serve still crashes our application, so can't use it on dev environment. If it doesn't appear on 1.7, would be nice to have the fix on 1.6.7

**Comment by DzianisUkrainets** [comment]
so which version of angular/cli does work for angular 5 ng build --prod ?

**Comment by imgregduh** [comment]
Angular 4

I am having this problem only with `ng build --prod` also 

@Londovir Concerning TFS build definitions,  I had to toggle `ContinueOnError `option under control options inside the task.  Downside to this, it will continue on ANY error though it will be recorded in the logs that it had errors.  I would think you run your application locally first before deploying, so build commands shouldn't usually have a problem.  This is only a bandaid and like all injuries if you bleed a lot 1 bandaid wont be enough
ref: [msdn](https://msdn.microsoft.com/en-us/library/ms171484.aspx)

**Comment by ubarevicius** [comment]
1.7.0-beta.3 release of the cli works fine for us, the error is gone and now it doesn't crash anymore at runtime while serving aot compiled websites.

**Comment by Londovir** [comment]
I may have to hold until 1.7.0 launches officially. Our team usually uses CLI in an ejected format, since we have custom tooling we need to add to the build chain, so in cases like this I'll need to use the CLI to create a shell project, eject it so I can see how they've changed the base webpack config files (likely removing cssnano since I believe that is the source of the deprecated "error"), and then update our customized webpack configs as needed.

**Comment by jfernandeze** [comment]
I'have the same issue

`"@angular/cli": "1.6.6",`
`    "bootstrap": "~4.0.0",`

Version of node: 8.9.4

ng build --prod    rc\assets\styles\style.scssNode#moveTo was Date: 2018-02-02T07:52:15.646Z 45% building modules 296/297 modules 1 active ...s\WebSpa\src\assets\styles\style.scssNode#moveTo was Date: 2018-02-02T07:52:15.646Z

**Comment by thovo** [comment]
Hi everyone, I had the same problem with cli 1.6.7, is there any update how to fix this?

**Comment by Baklap4** [comment]
Same here with cli 1.6.8 running on node 9.4.0

**Comment by manolosky** [comment]
Same here:

Angular CLI: 1.6.8
Node: 8.9.0
OS: darwin x64
Angular: 5.2.5

"bootstrap": "4.0.0-beta.2"

I need help...

**Comment by Sibiraj-S** [comment]
@manolosky . Update angular-cli to v1.7.0.

**Comment by Sathishchary** [comment]
I am getting same issue after converting from angular 4.3.1 to 5.2.5.

**Comment by rag7s** [comment]
> C:\xampp\htdocs\profile>ng build --prod
>  10% building modules 5/5 modules 0 activeNode#moveTo was deprecated. Use Container#append.
Getting above issue please help me with this.

**Comment by dojchek** [comment]
Guys, you should just upgrade angular-cli to 1.7.0+ and the warning will be gone.

**Comment by Baklap4** [comment]
Can confirm this running 1.7.2 and it disappeared

**Comment by rag7s** [comment]
I found the solution for this '0% building modules 5/5 modules 0 activeNode#moveTo was deprecated. Use Container#append.'
Example: if css has classes like this  `.container{color:red;} .container{font-size:14px;}` then you will get this error means douplicate classes. 
It should be  `.container{color:red;font-size:14px;}` then it wont show this error.

Like this I checked enter my project and then fixed this error.
